### ISSUE TITLE
Channel Name for upload

### ISSUE THREAD

**Issue by wrgr** [neurodata/CAJAL#63]
@alexbaden - why is channel name required for upload as part of the RAMONObject?

If my ocp class ('oo') has a channel of 'anno', and my object ('vol') has a channel name of 'will',

when I execute:

`oo.createAnnotation(vol)` what happens?  Where should this data get posted?  I argue channel name is helpful for bookkeeping, but should be ignored/not required on upload.

**Comment by alexbaden** [comment]
Channel name is required to package the HDF5 file. What will happen when
you post is the HDF5 will have data packaged as /will/CUTOUT but the target
channel will be anno, so presumably OCP will fail to eat the data.

**Comment by wrgr** [comment]
seems annoying maybe?  we're specifying the same data twice, right?
seems like we could draw packaging from the ocp packaging info instead?

the original idea of the channel in the object was just to help keep track
of where the data came from, not as part of the upload.

good idea or bad idea?

Best,
Will

**Comment by alexbaden** [comment]
Good idea. I can implement, but won't get to it until next week. Anyone else who wants to take a stab is welcome! 

I make liberal use of the Matlab debugger, which will show you all environment variables as your function sees them as its running. So, somewhere in `oo.createAnnotation`, we need to either pass the channel name to an HDF5 write function or access the appropriate field of the `oo` object (`this`) .

**Comment by alexbaden** [comment]
So the TODO is: change the query system to get the channel from oo, and display a deprecated message when q.setChannel is called (so we don't break people's existing queries)

**Comment by wrgr** [comment]
well, where q is an object.  So syn.setChannel should return a warning that
says, this field is ignored when posting data and is for client-side
bookkeeping only.  Or similar/
